File: java/warm-up-1.java

/*Warm up 1 Solutions */

// sleepIn

public boolean sleepIn(boolean weekday, boolean vacation) {
    if(!weekday &&  vacation) {
      return true;
    } else if(weekday && !vacation) {
      return false;
    } else if(!weekday && vacation) {
      return true;
    } else {
      return true;
    }
  }

// -------------------------------------------------------------------- //

// monkeyTrouble
public boolean monkeyTrouble(boolean aSmile, boolean bSmile) {
    if(!aSmile && !bSmile) {
      return true;
    } else if(aSmile && bSmile) {
      return true;
    } else {
      return false;
    }
  }
 
 // -------------------------------------------------------------------- //

// sumDouble
public int sumDouble(int a, int b) {
    if(a == b) {
      return (a + b) * 2;
    } else {
      return a + b;
    }
  }
 
 // -------------------------------------------------------------------- //

// diff21
public int diff21(int n) {
    if (n <= 21) {
      return 21 - n;
    } else {
      return (n - 21) * 2;
    }
  }

 // -------------------------------------------------------------------- //

// parrotTrouble
public boolean parrotTrouble(boolean talking, int hour) {
    if(talking && hour < 7) {
      return true;
    }
    if(talking && hour > 20) {
      return true;
    }
    else {
      return false;
    }
  }

// -------------------------------------------------------------------- //

// makes10
public boolean makes10(int a, int b) {
    return (a == 10 || b == 10 || a + b == 10);
  }
  
// -------------------------------------------------------------------- //

// near100
public boolean nearHundred(int n) {
    return (90 <= n && n <= 110 || 190 <= n && n <= 210);
  }
  
// -------------------------------------------------------------------- //

 //posNeg
 public boolean posNeg(int a, int b, boolean negative) {
    if (negative) {
      return (a < 0 && b < 0);
    }
    else {
      return ((a < 0 && b > 0) || (a > 0 && b < 0));
    }
  } 

File: python/warmup1.py
"""Codingbat Warmup-1 solutions

Practise solving programming challenges at codingbat.com
"""

# sleep_in
def sleep_in(weekday, vacation):
  if(not weekday and not vacation):
    return True
  elif(weekday and not vacation):
    return False
  elif(not weekday and vacation):
    return True
  else:
    return True

# monkey trouble
def monkey_trouble(a_smile, b_smile):
  return (a_smile == b_smile)

# sum_double
def sum_double(a, b):
  if a == b:
    return sum((a,b)) * 2
  return sum((a,b))

# dif21
def diff21(n):
  diff = abs(21 - n)
  if(n > 21):
    return diff * 2
  return diff

# parrot_trouble
def parrot_trouble(talking, hour):
  return (talking and (hour < 7 or hour > 20))

# makes10
def makes10(a, b):
  return (a==10 or b==10 or a + b == 10)

# near_hundred
def near_hundred(n):
  return ((abs(100 - n) <= 10) or (abs(200 - n) <= 10))

# pos_neg
def pos_neg(a, b, negative):
  if negative:
    return (a < 0 and b < 0)
  return ((a < 0 and b > 0) or (a > 0 and b < 0))

# not_string
def not_string(str):
  if(str.startswith('not')):
    return str
  return ('not ' + str)

# missing_char
def missing_char(str, n):
    new_string = ''
    new_string += str.replace(str[n], '')
    return new_string

# front_back
def front_back(str):
  if len(str) <= 1:
    return str
  return str[len(str)-1] + str[1:len(str)-1] + str[0]

# front3
def front3(str):
  if len(str) <= 3:
    return str * 3
  return str[ : 3] * 3